Fix/2675 docs tabs

kwongz opened the pull request:
### Description


https://github.com/user-attachments/assets/08f3356a-5213-461b-93fd-133ac4faf74f



**Notion Docs insturctions**
https://www.notion.so/Docs-Doc-Tabs-12a5287a4ba08046b7f1cef7f7b03eae

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- ~~[ ] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)~~
- [x] I have added to the docs where applicable
- ~~[ ] I have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

kwongz commented:
fixes #2675

archiewood commented:
This will be great!

Couple of suggestions in loom

mcrascal commented:
This is cool. Just looking at it in the preview docs: 

- the vertical slide transition currently fires on navigation to the page
- text-xs is way too small for this imo 
- needs more margin top, probably less margin bottom. Right now in a page with lots of these, it's not clear what sections it's denoting. 
- Need a horizontal bottom border for the width of the page
- Possibly need something wrapping the content which is inside the tabs

mcrascal commented on sites/docs/components/DocTab.svelte:
```diff
@@ -0,0 +1,93 @@
+				on:click={() => setTab(tab, index)}
+				bind:this={tabButtons[index]}
+			>
+				{toCapitilize(tab)}
```
Could we consider using the `capitalize` class instead of introducing a new utils function for this?

mcrascal commented on sites/docs/components/DocTab.svelte:
```diff
@@ -0,0 +1,93 @@
+		bind:this={tab}
+		style="height: {tabContentHeight};"
+		class="transition-height overflow-hidden ease-in-out duration-300"
+	>
```
Why isn't svelte `transition:slide|local` on the child divs suitable appropriate here?

kwongz replied:
Currently playing around with svelte transitions to see possible solutions.

Originally i chose to transition height because the default slot with the code grows in height after initially rendering on the page, causing the rest of the page to grow downwards (Giving a bad delayed jump effect on the content). This was handled smoothly by the height observers to detect and transition any changes in height of the div on tab change.

I think with the non svelte transitions we can achieve a lot more custom and fine tuned animations/transitions. But happy to explore other avenues.

kwongz commented:
@mcrascal 

1. Updated Accordion and Slider to DocTabs
2. Switched to svelte animations (still working on the smooth transition w/ svelte transitions

archiewood commented:
added a couple of suggestions on styling, feel free to revert if they feel less good.
- back to font-sans
- more padding around the preview
- adjust preview border colour and shadow to match code

archiewood commented:
The svelte slide animation is still too janky IMO.

![CleanShot 2024-10-31 at 11 27 37](https://github.com/user-attachments/assets/59fea7fc-59ed-4c61-bfbd-45cb56ed75db)

code -> preview is fine
but 
preview -> code has a weird jump at the end

archiewood commented:
on testing further, this seems related to the CodeBlock, not the transition. 

We can fix that separately

kwongz commented:
@archiewood I found a solution to the code --> preview, the prism loads in async, adding a margin after they lifecycle methods giving it the height 'jank'

By adding the exact margin as a tailwind style to the div container we avoid this height jank.

kwongz commented:
On Another note,

Current svelte transition unmount and mounts the preview content or code content. For pages like DataTable, where the DataTable get remounted, it can really slow down the page, as well as the transitions between the tabs.

A solution I'm looking into is doing CSS transitions and instead of unmounting the component, we can render everything and only make visible the content based on the tab chosen.

archiewood commented:
@kwongz the margin jank should have been fixed in #2706

kwongz commented:
@archiewood @mcrascal 

Added Custom CSS transitions to handle rendering issues with svelte transitions (unmounting/remounting components). Causing particular issues with datatables page.

The margin and padding layout is consistent across the preview and code block, but individual components such as Area Chart can appear to have more spacing due to margin/padding within the component itself.

I think its a pretty suddle difference that users won't notice immediately when they are consistent across components on the same doc page. But I thought we could introduce a standard margin for all components to create a more uniform look in the future.

_preview container margin_

![image](https://github.com/user-attachments/assets/7926d322-fe42-40ba-aa73-e50a623d9456)

_Component margin_

![image](https://github.com/user-attachments/assets/fa3747c5-28d5-42da-a60e-64bb48ef50f2)

__slider spacing not good__

![image](https://github.com/user-attachments/assets/62d961d8-e515-459c-b79e-a94ed63e3981)

archiewood commented on sites/docs/components/Pedistal.svelte:
is this component used

kwongz replied:
Pedistal was our original solution that had a cleaner transition, I left it out commented to come back to at another time

mcrascal reviewed (changes requested):
This seems close, but not quite there for something people will be using constantly in the docs. Prob worth a quick discussion. 

1. Transitioning this horizontally seems more annoying than useful. It doesn't remove the vertical jank, which I believe was the original goal of including a transition, but it adds quite a lot of time between seeing the output and the code. I think no transition between output and code would be preferable to the horizontal movement. 
2.  If we stay with the horizontal transition, you'd expect a carrousel type of movement (the preview is getting pushed off the screen by the code, and vice versa). This has the original object disappearing and then a new one sliding in. 
3. Focus rings on the buttons need styling 
4. I still find xs very small for something you're clicking all the time.

kwongz commented:
removed hover, added pointer, matched dropdown focus state

![image](https://github.com/user-attachments/assets/afcdb99f-a12c-45bc-bc21-15a425be9f27)

![image](https://github.com/user-attachments/assets/19eb17ed-554a-4b66-845f-285abc55856c)
